manifest-spec.md

```diff
@@ -1,37 +1,140 @@
-2. Following from that, we will likely store the manifest on some distributed protocol like IPFS or Swarm (although this is jumping ahead in the discussion).
-3. Since our off chain actors are primarily software, the message format needs to be easily readable by most programming languages.
-4. That said, it's likely human actors will write the manifest, meaning it needs to be in a format humans can easily understand and comprehend.
-5. Contrasting to the above, we need a format that takes up the least amount of space possible so that it costs the least to make available
+1. Package manifests will be consumed by package managers.
+2. Package manifests will be stored off-chain.
 
 
-## Data to be stored within the manifest
+## Use Cases
 
+The following use cases were used to guide the specification.
+
+1. Code with zero deployed instances.
+    1. Reusable base contracts.
+2. Code with one or more addresses.
+    1. Libraries: Installation of the package involves having an importable interface for the library and linking against the proper address during deployment.
+    2. Contracts: Installation of the package involves having an importable interface and some manner of templating in the proper addresses into the local source code prior to compilation as well as other non-contract code that may interact with the installed contract.
+
+
+It is worth pointing out that the *Library* and *Contract* use cases are functionally the same.  Once [Solidity issue #242](https://github.com/ethereum/solidity/issues/242) has been addressed this distinction may no longer need to exist.
+
+## Manifest format
+
+The canoonical format for the package manifest is a JSON document containing a
+single JSON object.  
+
+
+## Manifest filename
+convention is to name this document `epm.json` which is short for *'ethereum
+package manifest'*.
+
+## Manifest Data Overview
+
+The following fields are defined as data that *may* be included in a package
+manifest.  Custom fields should be prefixed with `x-` to prevent collision with
+new fields introduced in future versions of the specification.
+
+- manifest version
 - package name
 - author
-- version (preferably [semver](http://semver.org/))
+- version
 - description
 - contract metadata
   - abi
   - unlinked binary
-  - deployed address if applicable
-  - name/address pairs for linked libraries
+  - deployed addresses
+  - linked libraries
   - compiler version for bytecode verification
-  - key/value pairs of the sha3 hash of events this contract can emit with the abi definition of that event (i.e., the contract metadata should contain information for all events that can be triggered by this contract, including those triggered by linked libraries)
 - external project URIs (homepage, repository uri, etc.)
 - dependencies (i.e., references to other packages and versions)
-- updated time (likely added by the packager)
 
```

Comment:
My guess is external project URIs, which could be links to IPFS hashes, but we could include some other construct.